src/ensembl/src/content/app/entity-viewer/components/feature-length-axis/featureLengthAxisHelper.ts

```diff
@@ -0,0 +1,94 @@
+import { ScaleLinear } from 'd3';
+
+export const getTicks = (scale: ScaleLinear<number, number>) => {
+  // use d3 scale to get 'approximately' 10 ticks (exact number not guaranteed)
+  // which are "human-readable" (i.e. are multiples of powers of 10)
+  // and are guaranteed to fall within the scale's domain
+  let ticks = scale.ticks();
+  const length = scale.domain()[1]; // get back the initial length value on which the scale is based
+  const step = ticks[1] - ticks[0];
+
+  // choose only the "important" ticks for labelling
+  const exponent = Math.floor(Math.log10(length));
+  const powerOfTen = 10 ** exponent; // e.g. 100, 1000, 10000, etc.
+
+  if (length >= powerOfTen && length < powerOfTen + step) {
+    return handleLengthAsPowerOfTen(ticks, powerOfTen, step, length);
+  }
+
+  ticks = ticks.filter((number) => {
+    // do not add a tick in the end of the ruler (it is handled specially)
+    // and throw away all the possible 'inelegant' intermediate ticks, such as 50, etc.
+    return number !== length && number % powerOfTen === 0;
+  });
+
+  let labelledTicks = getLabelledTicks(ticks, powerOfTen, length);
+
+  if (!labelledTicks.length) {
+    // let's have at least one label, roughly in the middle of the ruler
+    const halvedPowerOfTen = powerOfTen / 2;
+    ticks = [...ticks, halvedPowerOfTen].sort();
+    labelledTicks = [halvedPowerOfTen];
+  }
+
+  return {
+    ticks,
+    labelledTicks
+  };
+};
+
+const handleLengthAsPowerOfTen = (
+  ticks: number[],
+  powerOfTen: number,
+  step: number,
+  totalLength: number
+) => {
+  ticks = ticks.filter((number) => {
+    if (number === powerOfTen) {
```

Comment:
Can this ever be true for any ticks other than the last one?

Reply:
I don't think so. It was definitely intended to check the last tick; so I updated the code to make it explicit.